### PR TITLE
Convert incoming DateTime strings to a local DateTime, and vice versa

### PR DESCRIPTION
**Pull request by bamcdole**

### CODE REVIEW

File: src/components/d2l-wizard-manager.js [hunk 1]
```diff
@@ -269,10 +270,19 @@ class WizardManager extends LocalizeMixin(LitElement) {
...
 
 	async _saveSchedule() {
+
+		// Copy the schedule cache we want to send, because we have to convert our local DateTime objects to UTC DateTime strings
+		// If the save fails, we don't want to keep the converted DateTimes in our cache
+		const lastRunTimeStr = getUTCDateTimeStringFromLocalDateTime(this.cachedSchedule.lastRunTime);
+		const nextRunTimeStr = getUTCDateTimeStringFromLocalDateTime(this.cachedSchedule.nextRunTime);
+		const copiedSchedule = JSON.parse(JSON.stringify(this.cachedSchedule));
+		copiedSchedule.lastRunTime = lastRunTimeStr;
+		copiedSchedule.nextRunTime = nextRunTimeStr;
```
Comment (bamcdole) [inline src/components/d2l-wizard-manager.js:280]: This is what I wanted to integration test the most.
If a fail saves, we need to make sure that our cachedSchedule still has Local DateTime objects, as opposed to UTC DateTime strings.

File: src/dateTime.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+	Logic for UTC -> Local DateTime conversion goes here
+	We are assuming the LMS will give us a UTC DateTime string in the ISO 8601 format (ex: 1994-11-05T13:15:30Z)
+	Returns a Javascript Date object
+*/
+export function getLocalDateTimeFromUTCDateTimeString(utcDateTimeStr) {
+	const localDateObj = new Date(utcDateTimeStr);
```
Comment (Syeda-Azim) [inline src/dateTime.js:7]: LastRunTime is nullable so we need to check for null or undefined dateTime.

File: src/components/d2l-schedule-logs.js [hunk 3]
```diff
@@ -173,6 +174,13 @@ class ScheduleLogs extends LocalizeMixin(LitElement) {
 		this.isQuerying = true;
 
 		const logs = await this.scheduleLogsService.getLogs(this.scheduleId, this.pageNumber, this.pageSize);
+
+		// UTC -> Local DateTime conversion
+		logs.forEach(log => {
+			log.runDate = getLocalDateTimeFromUTCDateTimeString(log.runDate);
+			log.endDate = getLocalDateTimeFromUTCDateTimeString(log.endDate);
```
Comment (lweber-d2l) [inline src/components/d2l-schedule-logs.js:181]: Here, `log.runDate` and `log.endDate` originally stored UTC DateTime strings.  After these calls, they will contain Date objects - I guess this is ok in JavaScript.

File: src/dateTime.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	return localDateObj;
+}
+
+/*
+	Logic for Local DateTime -> UTC conversion goes here
+	We are assuming that our frontend code will manipulate Javascript Date objects
+	Returns a UTC DateTime string, in the ISO 8601 format (ex: 1994-11-05T13:15:30Z)
+*/
+export function getUTCDateTimeStringFromLocalDateTime(localDateObj) {
+	const utcDateTimeStr = localDateObj.toISOString();
```
Comment (Syeda-Azim) [inline src/dateTime.js:17]: Check for null or undefined dateTime before conversion.

File: test/d2l-manage-schedules.test.js [hunk 5]
```diff
@@ -60,28 +60,38 @@ describe('d2l-manage-schedules', () => {
...
+				nextRunTime: '2020-10-20T15:00:00.000Z'
 			};
 
 			const el = await setFixtureSchedules(defaultFixture, [testSchedule]);
 
 			const rows = el.shadowRoot.querySelectorAll('tbody > tr');
 			expect(rows.length).to.equal(1);
 			const rowData = rows[0].querySelectorAll('td');
+
+			// <th>${ this.localize('scheduleName') }</th>
```
Comment (lweber-d2l) [inline test/d2l-manage-schedules.test.js:83]: Did you intend to include this commented-out block?  Our coding standard discourages checking in commented-out code.

Comment (bamcdole) [reply test/d2l-manage-schedules.test.js:83]: I did not. Will remove.

File: src/components/d2l-configure-schedule.js [hunk 6]
```diff
@@ -114,7 +114,7 @@ class ConfigureSchedule extends LocalizeMixin(LitElement) {
 	constructor() {
 		super();
 
-		const nowDate = formatDate(new Date(Date.now()), { format: 'yyyy-MM-dd' });
+		const nowDate = new Date(Date.now()).toString();
```
Comment (bamcdole) [inline src/components/d2l-configure-schedule.js:117]: We now expect a full DateTime string to be passed into this component

**Comment by IhabAlkabary** [comment]
> /test2

You need to retrigger this from jenkins now

**Comment by lweber-d2l** [comment]
> > /test2
> 
> You need to retrigger this from jenkins now

May be helpful: How to rerun CI integration tests: https://d2l.slack.com/archives/C0RTNDPC2/p1605883237365000

File: src/components/d2l-configure-schedule.js [hunk 7]
```diff
@@ -185,10 +194,17 @@ class ConfigureSchedule extends LocalizeMixin(LitElement) {
 	}
 
 	_commitChanges() {
+
+		// This is a bit ugly. Unfortunately, none of the Core Helpers can properly parse theDatePicker component output into a Date object
+		const startDateSplit = this.startDate.split('-');
+		const startDateObj = new Date(startDateSplit[0], startDateSplit[1] - 1, startDateSplit[2]);
```
Comment (Syeda-Azim) [inline src/components/d2l-configure-schedule.js:200]: Why are we subtracting 1 from month?

**Comment by JeremyWinkler** [comment]
> > > /test2
> > 
> > 
> > You need to retrigger this from jenkins now
> 
> May be helpful: How to rerun CI integration tests: https://d2l.slack.com/archives/C0RTNDPC2/p1605883237365000

That is for the LMS repo, in non LMS repos, go `Show all checks` -> `Details` -> `Re-run jobs`(Right side) -> `Re-run all jobs`
Looks like its passing now, but for future.

**Review by Syeda-Azim — APPROVED** [review]

**Pull request merged**